src/http_server.c

```diff
@@ -189,6 +189,17 @@ static void http_server_netconn_resp_file(struct netconn *conn, const char *file
 	netconn_write(conn, file_content, file_len, NETCONN_NOCOPY);
 }
 
+static void http_server_netconn_resp_metrics(struct netconn *conn) {
+	char* metrics = ruuvi_get_metrics();
```

Comment:
Preferably allocate the buffer here at the same scope where it is freed and pass it to function as a parameter.

Reply:
Since we don't know the required size for 'metrics' we can't allocate memory here. In all other places the same pattern is used, so it's ok for now, but I'll think how to improve it.